pkg/prometheus/promcfg.go

```diff
@@ -116,6 +116,34 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *v1.TLSConfig) yaml.M
 	return cfg
 }
 
+func addSafeAuthorizationToYaml(cfg yaml.MapSlice, version semver.Version, assetStoreKey string, store *assets.Store, auth *v1.SafeAuthorization) yaml.MapSlice {
+	if auth == nil || !version.GTE(semver.MustParse("2.26.0")) {
+		return cfg
+	}
+	authCfg := yaml.MapSlice{}
+	if auth.Type == "" {
+		auth.Type = "Bearer"
+	}
+	authCfg = append(authCfg, yaml.MapItem{Key: "type", Value: strings.TrimSpace(auth.Type)})
+	if auth.Credentials != nil {
+		if s, ok := store.TokenAssets[assetStoreKey]; ok {
+			authCfg = append(authCfg, yaml.MapItem{Key: "credentials", Value: s})
+		}
+	}
+	return append(cfg, yaml.MapItem{Key: "authorization", Value: authCfg})
+}
+
+func addAuthorizationToYaml(cfg yaml.MapSlice, version semver.Version, assetStoreKey string, store *assets.Store, auth *v1.Authorization) yaml.MapSlice {
+	if auth == nil || !version.GTE(semver.MustParse("2.26.0")) {
+		return cfg
+	}
+	authCfg := addSafeAuthorizationToYaml(yaml.MapSlice{}, version, assetStoreKey, store, &auth.SafeAuthorization)[0].Value.(yaml.MapSlice)
```

Comment:
it's a bit hard to grasp :) I don't have a better suggestion than type casting though

Reply:
Agreed, I implemented this like `addTLStoYaml`, but its not straight forward to understand. Will add a comment calling this out.